### PR TITLE
(review1): the first review

### PR DESCRIPTION
**Pull request by jwkanggist**
- 한글 주석을 달기위한 인코딩 추가
- 경로 문제로 동작하지 않음
-  파이썬에서 경로를 줄떄는 getcwd()를 사용해서 절대경로로 주는게 좋음 
- 지금 소스트리 구조에서 data/mnist 를 어디에 위치 시켜야 하는지 명확하지 않음
- 이후에 다시 리뷰 예정
- 1차 리뷰 종료
- 본 PR을 머지 하지 않아도 됨

### CODE REVIEW

**Comment by tucan9389** [comment]
```
TF Study 1
├ data
  ├ mnist
    ├ 0
    ├ 1
    ├ 2
    ...
    ├ 9
```
이런식으로 폴더 계층을 만들었구요. 0부터 9까지 폴더 안에 손글씨 이미지가 들어가있습니다.
데이터는 아래 url에 압축해서 올려놧어요.
https://drive.google.com/file/d/1jK3PBtYyg57gUPifeFnkj18SAAhyjrOE/view?usp=sharing

그리고 코드상에서 train data와 test data를 구별하도록 해놨습니다.

File: TF Study 1/data_loader.py [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
+# ------------
+# 한글 주석을 달려면 위 전처리기 필요
+
 # 데이터를 빠르게 불러와 메모리에 로드
 
 import os
 import numpy as np
 
 # 경로로부터 path와 label 조합의 numpy.ndarray
-def my_read_dataset_path_and_label(path = 'data/mnist/'):
```
Comment (jwkanggist) [inline TF Study 1/data_loader.py:7]: 경로는 명시적으로 표시해주는게 좋은 path='./data/mnist' 추천

File: TF Study 1/data_loader.py [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
-    dataset = []
+def my_read_dataset_path_and_label(path = './data/mnist/'):
     labels = [ name for name in os.listdir(path) if os.path.isdir(os.path.join(path, name)) ]
+
     for label in labels:
         image_filenames = os.listdir(os.path.join(path, label))
         dataset.extend([ [label + "/" + image_filename, label] for image_filename in image_filenames ])
```
Comment (jwkanggist) [inline TF Study 1/data_loader.py:17]: 저는 여기서 도영님꼐서 s [ "label / image_filename", label] 포맷으로 표현하신게 이해가 잘 안가는데요. label정보가 중복된다고 생각합니다. 다른 의미가 있으신가요 ?

File: TF Study 1/input_ops.py [hunk 3]
```diff
@@ -1,3 +1,8 @@
+#-*- coding: utf-8 -*-
+#! /usr/bin/env python
+# ------------
+# 한글 주석을 달려면 위 전처리기 필요
+
 # 사용 모델 인풋 형식에 맞게 변환 (여러 데이터를 쉽게 바꿔끼고 섞을 수 있어야한다.)
 
 from sklearn.model_selection import train_test_split
```
Comment (jwkanggist) [inline TF Study 1/input_ops.py:8]: train_test_split(X, y, test_size = 0.2)는 sklearn함수 같네요.
제가 잘 모르는 데 어떤 역확을 하는 함수 인지 간략하게 설명해 주실수 있나요 ?

File: TF Study 1/data_manager.py [hunk 4]
```diff
@@ -1,4 +1,7 @@
-
+#-*- coding: utf-8 -*-
+#! /usr/bin/env python
+# ------------
+# 한글 주석을 달려면 위 전처리기 필요
 
 from data_loader import my_read_dataset_path_and_label
 from input_ops import my_train_test_split
```
Comment (jwkanggist) [inline TF Study 1/data_manager.py:7]: data_manager.py에서  아래 두부분은 my_train_test_split()안으로 들어가는게 날꺼 같습니다
        self.y_train = np.array([int(y) for y in self.y_train])
        self.y_test  = np.array([int(y) for y in self.y_test])

Comment (jwkanggist) [reply TF Study 1/data_manager.py:7]: my_train_test_split()에서 차라리 한번에 ndarray로 하는건 어떨까요 ?

y = np.array(y ,dtype = np.uint8)

Comment (jwkanggist) [reply TF Study 1/data_manager.py:7]: get_input_data()에서 
np.zeros( shape = () ) 라고 명시적으로 써주는게 좋습니다

Comment (jwkanggist) [reply TF Study 1/data_manager.py:7]: for i in range(self.batch_size):
            one_batch_X[i] = get_image_data(self.dataset_path + X[i], self.data_width, self.data_height, self.data_channel)

여기서 이런식으로 한장씩 뺴오는게 최선인지 다시 생각해 보시기 바랍니다. for문은  cost가 높아요 도영님

Comment (jwkanggist) [reply TF Study 1/data_manager.py:7]: prepare_data()와 같은 추상적인 함수명은 안쓰는게 좋습니다. 차라리 extract_data_from_png()가 낫겠네요

**Review by jwkanggist — COMMENTED** [review]
잘 하셨습니다 몇가지 코멘트 드렸구요 마지막으로 아래에 대해서 다시 생각해 보시기 바랍니다. 
수고하셨습니다 :-) 👍 
------------------------
이미지 파일로 부터 데이터를 효율적으로 추출하는 모듈인가?
중간에 cast / 중복 처리로 비효율은 발생하지 않는가?
MNIST외의 다른 데이터 셋에 적용할 수 있을 정도로 유연한가?
최신 api를 충분히 사용하고 있는가?

File: TF Study 1/data_loader.py (hunk 2 above)
Comment (tucan9389) [reply TF Study 1/data_loader.py:17]: 제가 다른 데이터셋을 많이 못봐서, 보통 어떤식으로 레이블을 가지고 있는지는 모르겠습니다만.. 
메모리상에 (경로, 레이블) 포맷으로 가지고 있어야 좀 더 다양한 데이터셋 폴더 구조를 대응할 수 있다고 생각했습니다. 제가 준비한 MNIST 데이터는 ./data/{label_name}/{image_name} 이런식이지만, 다른 데이터셋은 폴더에 레이블명이 명시되있지 않을 수도 있다고 생각했거든요.

File: TF Study 1/input_ops.py (hunk 3 above)
Comment (tucan9389) [reply TF Study 1/input_ops.py:8]: 예전에 들었던 ML 강좌에서 소개해준 sklearn 라이브러리인데요. `train_test_split` 함수에 데이터셋을 넣으면 학습 데이터와 테스트 데이터로  나뉘어 줍니다. 그 비율은 test_size로 나뉘구요. 디폴트로는 자동으로 데이터를 섞어서 나눠줘요.

File: TF Study 1/data_manager.py (hunk 4 above)
Comment (tucan9389) [reply TF Study 1/data_manager.py:7]: for문의 대안은 한번 고민해보겠습니다. 잘 안되면 참고자료 보구요~ 그래도 모르겠으면 나중에 여쭤볼게요.

Comment (tucan9389) [reply TF Study 1/data_manager.py:7]: 리뷰하기 힘드셔서 그렇게 말씀하신거죠? 
추상적이지 않은 함수명에대한 제안도 좀 더 고민해보겠습니다. 

제가 iOS 개발할때 최종적으로 불리는 함수는 추상적으로 작명하는 편이었거든요. 제일 고민하게 만드는 부분이네요 ㅎㅎㅎ

File: TF Study 1/data_loader.py (hunk 2 above)
Comment (jwkanggist) [reply TF Study 1/data_loader.py:17]: 그럴수도 있지만 
보통 이런경우는 파일명의 규칙성이 불명확해서 파씽하기가 어렵게 만들 수 있습니다.

File: TF Study 1/input_ops.py (hunk 3 above)
Comment (jwkanggist) [reply TF Study 1/input_ops.py:8]: 네 잘았겠습니다 👍

File: TF Study 1/data_manager.py (hunk 4 above)
Comment (jwkanggist) [reply TF Study 1/data_manager.py:7]: 추상화가 필요한 api명의 경우 그렇케 하는게 좋죠. 아닌경우 구체적인데 낫습니다. 

prepare_data()가 extract_data_from_png()을 wrapping하고 있는 형태가 일반적이 되겠네요
